p2p/conn/secret_connection.go

```diff
@@ -124,14 +123,18 @@ func (sc *SecretConnection) Write(data []byte) (n int, err error) {
 		binary.BigEndian.PutUint32(frame, uint32(chunkLength))
 		copy(frame[dataLenSize:], chunk)
 
+		aead, err := xchacha20poly1305.New(sc.shrSecret[:])
+		if err != nil {
+			return n, errors.New("Invalid SecretConnection Key")
+		}
 		// encrypt the frame
-		var sealedFrame = make([]byte, sealedFrameSize)
-		secretbox.Seal(sealedFrame[:0], frame, sc.sendNonce, sc.shrSecret)
+		var sealedFrame = make([]byte, aead.Overhead()+totalFrameSize)
```

Comment:
is there a way to get this overhead as a constant so we can bring back `sealedFrameSize` ?

Reply:
The AEAD api only makes it available as a function. The AEAD interface only exposes it as function and it's a go standard library interface

Reply:
For ChaCha20Poly1305 it will always be 16 (the length of a Poly1305 tag): https://github.com/golang/crypto/blob/master/chacha20poly1305/chacha20poly1305.go#L46